python/ray/_raylet.pyx

```diff
@@ -643,54 +654,66 @@ cdef class CoreWorker:
                      resources,
                      placement_resources):
         cdef:
-            ActorHandle actor_handle = ActorHandle.__new__(ActorHandle)
+            unique_ptr[CActorHandle] actor_handle
```

Comment:
Given that we're not returning the actor handle, why not just create it in the C++ code and not expose it at all here?

Reply:
I would've preferred to do that, but unfortunately the method that creates the `ActorHandle is on the `CoreWorkerTaskInterface`, which doesn't have access to the `CoreWorker`, and the `CoreWorker` is where the handle needs to be stored eventually. So I chose to keep the current structure and give the `CoreWorker` the handle here instead.

In the future, if we merge `CoreWorkerTaskInterface` and `CoreWorker`, I think we should do what you suggested.